trio/_path.py

```diff
@@ -142,7 +142,7 @@ def __repr__(self):
 
     def __fspath__(self):
         try:
-            return self._wrapped.__fspath__()
+            return fspath(self._wrapped)
         # python3.5 compat
         except AttributeError:  # pragma: no cover
             return str(self)
```

Comment:
Tests are failing, and I think it's because this fallback isn't being hit anymore.

Reply:
You are right. I think the issue though is that the PEP implementation does not work with pre Python 3.6 pathlib objects.

I just pushed a new commit where I added that check on `fspath()` itself. If it works, then this workaround can probably be removed.